### DISCUSSION TITLE
User/Groups: Question about "Organization" and "Organizational id"

### DISCUSSION THREAD

**Discussion by MatKDA** [leihs/leihs#1292] [Q&A / Community Support]
Hello,

I’m currently trying out the user/group management. However, I’m a little bit lost with the fields “Organization” and “Organizational ID”.

First, the behavior is not the same for users and groups: When creating a user I have to fill out the “Organization” field but when creating a group the field is automatically set to the recommended setting (“leihs-local”), when I don’t enter anything – is this the intended behavior?

Second, the help text to “Organizational ID” confuses me: “The combination of organization and org_id must be unique accross all entities.” I don’t know what “entity” refers to, but I can create multiple users/groups with “Organization” set to “leihs-local” and “Organizational ID” left empty. In this case, however, organization and org_id are not unique across all entities.

Is it fine to set “Organization” to “leihs-local” and leave “Organizational ID” empty or will this lead to problems? Could you perhaps give example use cases for these fields?

Thanks in advance!
Regards,
Matthias

**Comment by DrTom** [comment]
The most common use case is to differentiate between entities (an entity can be a user or a group) which are managed `locally` vs those managed automatically by some sort of import or sync. 

It is possible to go further and differentiate users from multiple tenants (rare but actually there are leihs instances). 

The default for both the users and the groups table is (now) `local`. 

> “The combination of organization and org_id must be unique across all entities.”
Well, there is an unique index. So this is correct but it also allows the `org_id` to be `null` in which case it is strictly speaking not unique. This is the intended behavior. Instances where only manual managed users are present usually just use `local` in the org field and do not bother with `org_id`.
